sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/ConversationClient.java

```diff
@@ -236,4 +241,106 @@ public GetCallRecordingStateResult getRecordingState(String conversationId, Stri
+            .block();
+    }
+
+    /**
+     * Download the content located in {@code endpoint} into a file marked by {@code path}.
+     * This download will be done using parallel workers.
+     * @param path - File location.
+     * @param endpoint - ACS URL where the content is located.
+     * @param range - An optional {@link HttpRange} value containing the range of bytes to download. If missing,
+     *                  the whole content will be downloaded.
+     * @param parallelDownloadOptions - an optional {@link ParallelDownloadOptions} object to modify how the parallel
+     *                               download will work.
+     * @param override - True to override the file if it exists.
+     * @return {@link Void}
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Void downloadTo(String path, URI endpoint, HttpRange range,
+                                                     ParallelDownloadOptions parallelDownloadOptions,
+                                                     boolean override) {
+        return conversationAsyncClient.downloadTo(path, endpoint, range,
+            parallelDownloadOptions, override).block();
+    }
+
+    /**
+     * Download the content located in {@code endpoint} into a file marked by {@code path}.
+     * This download will be done using parallel workers.
+     * @param path - File location.
+     * @param endpoint - ACS URL where the content is located.
+     * @param range - An optional {@link HttpRange} value containing the range of bytes to download. If missing,
+     *                  the whole content will be downloaded.
+     * @param parallelDownloadOptions - an optional {@link ParallelDownloadOptions} object to modify how the parallel
+     *                               download will work.
+     * @param override - True to override the file if it exists.
+     * @param context A {@link Context} representing the request context.
+     * @return Response containing the http response information from the download.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<Void> downloadToWithResponse(String path, URI endpoint, HttpRange range,
+                                                     ParallelDownloadOptions parallelDownloadOptions,
+                                                     boolean override, Context context) {
+        return conversationAsyncClient.downloadToWithResponse(path, endpoint, range,
+            parallelDownloadOptions, override, context).block();
+    }
+
+    /**
+     * Download the content located in {@code endpoint}, e.g. Recording video, from the ACS endpoint passed as
+     * parameter.
+     * @param endpoint - ACS URL where the content is located.
+     * @return The byte stream of the requested content.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Flux<ByteBuffer> downloadStreaming(URI endpoint) {
```

Comment:
Generally we don't return reactive types (`Mono`, `Flux`, `PollerFlux`, `PagedFlux`) in the synchronous client. Downloading functionality doesn't use the same parity as simpler CRUD client APIs.

Reply:
What should I return instead? Store the flux content into a `ByteArrayOutputStream` and return that instead?

Reply:
I'd recommend removing this completely and following a pattern similar to `azure-storage-blob` where the asynchronous client has a `download` API which returns a `Flux<ByteBuffer>` but the synchronous client has what you have in `downloadTo` which takes an `OutputStream`.

https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java#L591

https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java#L855